lib/logproto/logproto-buffered-server.c

```diff
@@ -624,7 +624,7 @@ log_proto_buffered_server_fetch_from_buffer(LogProtoBufferedServer *self, const
       goto exit;
     }
 
-  success = self->fetch_from_buffer(self, buffer_start, buffer_bytes, msg, msg_len);
+  success = self->fetch_from_buffer(self, &buffer_start, buffer_bytes, msg, msg_len);
```

Comment:
why do we need to propagate the changes to buffer_start back here? This is a local variable which is ignored after fetch_from_buffer() returns. I am still reading the code to see what I am missing here.

Reply:
I've removed this part.